Clean up and NEWS.md for v1.0.0

k-doering-NOAA opened the pull request:
This adds the NEWS.md info for v1.0.0, and also includes some commits that are minor cleanup steps of documentation.

@Bai-Li-NOAA - I think it would be great to have your help reviewing the release notes, since you also contributed code! Would you be able to review it within a week? If not, though, I can ask for help from another NSAP member.

Feel free to note if elements should be changed, or you can push commits directly to the branch, if you would like.

Once this is merged in, we can make sure doc and style runs and merge it in if needed; then tag and officially release v1.0.0!

<!---
Thanks for opening a PR. This commented text will **NOT** appear in the final PR. Toggle between Write and Preview to see what your PR will look like without the comments.

The following individuals are available to review ghactions4r pull requests, so please select one of them as the reviewer and ask directly on the pull request (using `@mention`) if they are comfortable reviewing and provide a date you would like the review by.

 k-doering-NOAA Bai-Li-NOAA kellijohnson-NOAA Andrea-Havron-NOAA Schiano-NOAA msupernaw e-perl-NOAA ClaireGonzales-NOAA
-->

k-doering-NOAA committed:
first draft of NEWS for 1.0.0

k-doering-NOAA committed:
issue #234: suppress documentation of internal functions

k-doering-NOAA committed:
edit NEWS

k-doering-NOAA committed:
bump version

k-doering-NOAA committed:
documentation fixes

k-doering-NOAA committed:
devtools::document()

Bai-Li-NOAA commented:
@k-doering-NOAA, the plan sounds good. I will do my best to get it done by the end of Sep 30.

Bai-Li-NOAA commented on NEWS.md:
```diff
@@ -1,6 +1,26 @@
 
+# ghactions4r 1.0.0
+
+This is technically the first stable release of ghactions4r! The API will not change unexpectedly.
```
nitpick: Maybe spell out "API" the first time it's mentioned?

k-doering-NOAA replied:
I realized perhaps linking the semantic versioning website might be better than using "API" at all, for folks who want all the details. I think the "first stable release" captures the idea that the API won't change unexpectedly for users, without using jargon.

Bai-Li-NOAA commented on NEWS.md:
```diff
@@ -1,6 +1,26 @@
+
+## Major Changes
+
+* Allow style and doc to work on multiple branches and make PRs for each branch in https://github.com/nmfs-ost/ghactions4r/pull/193 
```
nitpick: I noticed “style and doc,” “doc and style,” and “doc-and-style” are all used in the release notes. Maybe choose one consistent form.

k-doering-NOAA replied:
Great point! I corrected this.

Bai-Li-NOAA reviewed (approved):
@k-doering-NOAA This is exciting! I have a couple of minor nitpick. Feel free to keep things as is if you prefer.

k-doering-NOAA commented:
> @k-doering-NOAA This is exciting! I have a couple of minor nitpick. Feel free to keep things as is if you prefer.

Thank you! I will make edits based on your comments today!

k-doering-NOAA committed:
changes to NEWS.md
use consistent terminology to refer to the doc-and-style workflow, fix spelling error.
Thanks @Bai-Li-NOAA for suggesting consistent doc-and-style workflow terminology.
